api/services/device_tags.go

```diff
@@ -62,24 +76,12 @@ func (s *service) UpdateDeviceTag(ctx context.Context, uid models.UID, tags []st
 		}
 
 		return l
-	}
-
-	if len(tags) > DeviceMaxTags {
```

Comment:
Same here.

The route check if the maximum number of allowed tags, and each one is valid.